scripts/zkai_native_block_boundary_pivot_selector_gate.py

```
@@ -0,0 +1,655 @@
+    writer = csv.DictWriter(output, fieldnames=ROW_COLUMNS, delimiter="\t", lineterminator="\n")
+    writer.writeheader()
+    for row in payload["routes"]:
+        writer.writerow({key: row[key] for key in ROW_COLUMNS})
+    return output.getvalue()
+
+
+def write_text(path: pathlib.Path, text: str) -> None:
+    if not path.is_absolute():
+        path = ROOT / path
+    if path.suffix not in {".json", ".tsv"}:
+        raise PivotSelectorError(f"unsupported output suffix: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)